directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py

```diff
@@ -0,0 +1,267 @@
+        open_item.connect("activate", lambda _: self.launch(directory.get_uri(), Gtk.get_current_event_time()))
+        menu.append(open_item)
+
+        term_image = Gtk.Image.new_from_icon_name("terminal", Gtk.IconSize.MENU)
+        term_item = self.new_image_menu_item("Open in Terminal", term_image)
+        term_item.connect("activate", lambda _: self.open_terminal_at_path(directory.get_path()))
+        menu.append(term_item)
+
+        menu.append(Gtk.SeparatorMenuItem.new())
+
+        # log(directory_uri)
+
+        iter = directory.enumerate_children('standard::*', Gio.FileQueryInfoFlags.NONE, None)
+
+        dirs = []
+        nondirs = []
+
+        info = iter.next_file(None)
+        while info != None:
+            if (not info.get_is_hidden() or self.show_hidden):     # // <-- skip hidden files
+                info.display_name = info.get_display_name()
+                info.content_type = info.get_content_type()
+                info.file = directory.get_child_for_display_name(info.display_name)
+                info.is_directory = (info.get_content_type() == "inode/directory")
+
+                if (info.is_directory):
+                    dirs.append(info)
+                else:
+                    nondirs.append(info)
+
+            info = iter.next_file(None)
+
+        dirs.sort(key = lambda info: info.display_name.lower())
+        nondirs.sort(key = lambda info: info.display_name.lower())
+
+        for info in dirs:
+            self.add_to_menu_from_gioinfo(menu, info)
+        for info in nondirs:
+            self.add_to_menu_from_gioinfo(menu, info)
+
+    def add_to_menu_from_gioinfo(self, menu, info):
+        display_text = info.display_name
+
+        icon = Gio.content_type_get_icon(info.content_type)
+        image = Gtk.Image.new_from_gicon(icon, Gtk.IconSize.MENU)
+
+        uri = info.file.get_uri()
+
+        item = self.new_image_menu_item(display_text, image)
+
+        if info.is_directory:
+            subMenu = self.create_subdirectory_submenu(uri)
+            item.set_submenu(subMenu)
+        else:
+            item.connect("activate", lambda _: self.launch(uri, Gtk.get_current_event_time()))
+
+        menu.append(item)
+
+    def create_subdirectory_submenu(self, uri):
+        subMenu = Gtk.Menu.new()
+        subMenu.set_reserve_toggle_size(False)
+
+        def f(_):
+            self.populate_menu_with_directory(subMenu, uri)
+            subMenu.show_all()
+
+        subMenu.connect("show", f)
+
+        subMenu.connect("hide", lambda _: self.destroy_all_children_later(subMenu))
+
+        return subMenu
+
+    def launch(self, uri, timestamp):
+        print(json.dumps({
+            "action": "launch_default_for_uri",
+            "uri": uri,
+            "timestamp": timestamp,
+        }))
+
+    def open_terminal_at_path(self, path):
+        print(json.dumps({
+            "action": "open_terminal_at_path",
+            "path": path,
+        }))
```

Comment:
Should these do more than print?

Reply:
#5453

Reply:
Not really, see https://github.com/linuxmint/cinnamon-spices-applets/pull/5453#pullrequestreview-1861465938